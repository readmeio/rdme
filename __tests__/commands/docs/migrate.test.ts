import { beforeAll, describe, expect, it } from 'vitest';

import Command from '../../../src/commands/docs/migrate.js';
import { type OclifOutput, runCommand } from '../../helpers/oclif.js';

describe('rdme docs migrate', () => {
  let run: (args?: string[]) => OclifOutput;

  beforeAll(() => {
    run = runCommand(Command);
  });

  it('should error out if no path is passed', async () => {
    const output = await run();
    expect(output).toMatchSnapshot();
  });

  it('should hide warning if `--hide-experimental-warning` flag is passed', async () => {
    const output = await run(['__tests__/__fixtures__/docs/new-docs', '--hide-experimental-warning']);
    expect(output).toMatchSnapshot();
  });

  it('should error out if no plugins are installed', async () => {
    const output = await run(['__tests__/__fixtures__/docs/new-docs']);
    expect(output).toMatchSnapshot();
  });

  it.todo('should load plugin and transform docs');
});
