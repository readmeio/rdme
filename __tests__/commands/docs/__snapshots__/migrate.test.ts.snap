// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rdme docs migrate > should error out if no path is passed 1`] = `
{
  "error": [Error: Missing 1 required arg:
path  Path to a local Markdown file or folder of Markdown files.
See more help with --help],
  "stderr": "",
  "stdout": "",
}
`;

exports[`rdme docs migrate > should error out if no plugins are installed 1`] = `
{
  "error": [Error: This command requires a valid migration plugin.],
  "stderr": " ›   Warning: This command is in an experimental alpha and is likely to change.
 ›    Use at your own risk!
- 🔍 Looking for Markdown files in the \`__tests__/__fixtures__/docs/new-docs\` directory...
✔ 🔍 Looking for Markdown files in the \`__tests__/__fixtures__/docs/new-docs\` directory... 1 file(s) found!
",
  "stdout": "",
}
`;

exports[`rdme docs migrate > should hide warning if \`--hide-experimental-warning\` flag is passed 1`] = `
{
  "error": [Error: This command requires a valid migration plugin.],
  "stderr": "- 🔍 Looking for Markdown files in the \`__tests__/__fixtures__/docs/new-docs\` directory...
✔ 🔍 Looking for Markdown files in the \`__tests__/__fixtures__/docs/new-docs\` directory... 1 file(s) found!
",
  "stdout": "",
}
`;
